python/skupper_router_internal/management/config.py

```diff
@@ -48,9 +48,11 @@ class Config:
     def __init__(
             self,
             filename: Optional[str] = None,
-            schema: QdSchema = QdSchema(),
+            schema: QdSchema = None,
             raw_json: bool = False
     ) -> None:
+        schema = schema or QdSchema()
```

Comment:
I want to understand the problem here. This class is instantiated in this file and the init is called once.  Is there a fear that QdSchema would be somehow manipulated and passed back in again. Could we just remove the schema parameter  and instantiate QdSchema inside the function.

Reply:
The fear is the usual "Mutable default arguments" gotcha, https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments, yes.

> Could we just remove the schema parameter and instantiate QdSchema inside the function.

That would work too. The constructor is never called with the `schema` parameter, the value is always defaulted.

Reply:
> The fear is the usual "Mutable default arguments" gotcha, https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments, yes.
> 
> > Could we just remove the schema parameter and instantiate QdSchema inside the function.
> 
> That would work too. The constructor is never called with the `schema` parameter, the value is always defaulted.

I prefer to just remove the schema parameter altogether and instantiate the schema inside the function. It is clean and easy to understand.